smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -70,7 +70,7 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
     protected fun needsResponseDecoder(test: HttpResponseTestCase): Boolean {
         var needsDecoder = false
         test.body.ifPresent { body ->
-            if (body.isNotBlank() && body.isNotEmpty() && test.bodyMediaType.isPresent) {
+            if (body.isNotBlank() && body.isNotEmpty()) {
```

Comment:
If my memory serves me well (it often doesn't), I think it was there before because we were it ( a long time ago) we were using it to figure out what decoder to use.  Since then, we have changed it to delegate to httpProtocolCustomizable, which means we no longer need to have a static map between `application/xml` (or `application/json`) to `XMLDecoder` (or `JSONDecoder`)

Reply:
oh you are so right!

Reply:
wow so glad you removed it more so now